### PR TITLE
[GDPR10] Log in as a minor

### LINKED ISSUE #5: [GDPR10] Log in as a minor
Reference: GDPR10

**Is your feature request related to a problem? Please describe.**
As a minor, once my account has been created for me by my guardian (see "Associated account for minors", ref: GDPR09), I would like to be able to log in.

**Describe the solution you'd like**
As a minor, I would like to log in with my e-mail address and password.

I would like to change my user password. As this user I will have access to minors spaces (see "Spaces with access by minors", ref: GDPR16).

As the access for minors depends on the context of each organisation, this development must be a module

**Describe alternatives you've considered**
Have an exclusive login path for minors.

**Does this issue could impact on users private data?**
Yes as it afects to minors data, it is particulary sensitive. 

**Acceptance criteria**
- [x] As a minor registered I can be able to log in
- [x] As a minor registered I can be able to change my password
- [x] As a minor registered I can be able to change my profile data, except for my birthday
- [x] As a minor registered, I can only receive private messages from other underage users.
- [x] As a minor registered I can't be able to access to authorizations page

### PR DESCRIPTION
**Pull request by fblupi**
Fixes #5 

With this PR, minor users won't be able to access the authorizations page.

- [x] As a minor registered I can be able to log in
- [x] As a minor registered I can be able to change my password
- [x] As a minor registered I can be able to change my profile data, except for my birthday
- [x] As a minor registered, I can only receive private messages from other underage users.
- [x] As a minor registered I can't be able to access to authorizations page

### CODE REVIEW

File: app/permissions/decidim/kids/permissions.rb [hunk 1]
```diff
@@ -18,6 +19,12 @@ def permissions
               allow!
             end
           end
+        when :authorizations
+          if !user.organization.minors_participation_enabled? || user.minor?
```
Comment (microstudi) [inline app/permissions/decidim/kids/permissions.rb]: In here if minors participation is disable, no one will be able to access the authorizations. In this case we should do nothing and let the default behaviour.

**Review by microstudi — CHANGES_REQUESTED** [review]
I see that this is for limiting certain accesses to minor (probably we will add cases in the future).
Not sure that disabling completely the authorization page is the way to go, but we can improve it in the future if needed.

I left you a comment, but, overall, this PR needs tests, probably in the `specs/permissions` folder.

File: app/controllers/concerns/decidim/kids/has_minors_permissions.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Kids
+    module HasMinorsPermissions
```
Comment (microstudi) [inline app/controllers/concerns/decidim/kids/has_minors_permissions.rb]:
```suggestion
    module HasDecidimKidsPermissions
```
as this is not exclusive for the minors, maybe a more generic name would be better

File: app/permissions/decidim/kids/permissions.rb [hunk 3]
```diff
@@ -6,27 +6,46 @@ class Permissions < Decidim::DefaultPermissions
...
+        authorizations_action?
+        conversation_action?
 
         permission_action
       end
 
+      def minor_accounts_action?
+        return unless permission_action.subject == :minor_accounts
+
+        if !user.organization.minors_participation_enabled? || user.minor? || !user.confirmed?
```
Comment (microstudi) [inline app/permissions/decidim/kids/permissions.rb:21]: I think that this line breaks access to authorizations an other parts if you disable the module in `/system`

**Review by microstudi — CHANGES_REQUESTED** [review]
Thanks @fblupi 
this looks ok but it has a problem if you disable the module in the `/system`, no person is allowed to access authorizations then.

Can we create a `spec/system` test to check the accesses?
we need to make sure when `organization.minors_participation_enabled?` is false that this permissions don't do anything and delegate everything to the normal decidim.

**Review by microstudi — COMMENTED** [review]
Hi @fblupi 
Everything looks good, good job!
I would be nice to have and additional integration test (spec/system) to check different accesses to the manipulated controllers. I feel that this kind of minor permission control will expand in the future and it would be nice to have a final test to expand.

i would tests these cases to start:

1. when the module is activated:
  1. If I'm a normal user I can access authorizations and start conversations with anyone exept minors
  2. If I'm a minor I cannot access authorizations and i cannot start conversations with adults (or can I?)
2. when the module is not activated
  1. If I'm a normal user I can access authorizations and start conversations with anyone, even if is a minor (as the module will still be activated it is possible to have former "minor users")
  2. If I'm a minor I can access authorizations and I can start a conversation with an adult (following the same hypothetical case of being a former minor)

**Comment by fblupi** [comment]
> Hi @fblupi Everything looks good, good job! I would be nice to have and additional integration test (spec/system) to check different accesses to the manipulated controllers. I feel that this kind of minor permission control will expand in the future and it would be nice to have a final test to expand.
> 
> i would tests these cases to start:
> 
> 1. when the module is activated:
> 2. If I'm a normal user I can access authorizations and start conversations with anyone exept minors
> 3. If I'm a minor I cannot access authorizations and i cannot start conversations with adults (or can I?)
> 4. when the module is not activated
> 5. If I'm a normal user I can access authorizations and start conversations with anyone, even if is a minor (as the module will still be activated it is possible to have former "minor users")
> 6. If I'm a minor I can access authorizations and I can start a conversation with an adult (following the same hypothetical case of being a former minor)

Hi @microstudi,
@apuertadelaguila has made the tests you requested

**Review by microstudi — APPROVED** [review]
Sorry for being late! this looks good now!

**Pull request merged**